### PR TITLE
solution and test for 5-core-get-api-articles

### PR DESCRIPTION
**Pull request by miors**
endpoint /api/articles
added solution and test
update test description

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -81,3 +82,36 @@ describe("GET /api/articles/:article_id", () => {
...
+
+describe("GET /api/articles", () => {
+  it("GET:200 should send all articles", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toHaveLength(13);
+        articles.forEach((article) => {
+          expect(Object.keys(article)).toHaveLength(8);
```
Comment (nc-niamh-smith) [inline __tests__/integration.test.js:94]: ^ I wouldn't worry about this expect - as you may increase or decrease the amount of keys it has! :)

File: __tests__/integration.test.js [hunk 2]
```diff
@@ -81,3 +82,36 @@ describe("GET /api/articles/:article_id", () => {
...
+
+  it("GET:200 should sort using created_at by default", () => {
+    return request(app)
+      .get("/api/articles")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
```
Comment (nc-niamh-smith) [inline __tests__/integration.test.js:115]: Great use of toBeSortedBy!

File: models/articles.model.js [hunk 3]
```diff
@@ -11,3 +11,14 @@ exports.fetchArticleByID = (article_id) => {
       return rows[0];
     });
 };
+
+exports.fetchAllArticles = () => {
+  return db
+    .query(
+      `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comment_id)::int AS comment_count FROM 
+      articles LEFT JOIN comments ON articles.article_id  = comments.article_id GROUP BY articles.article_id ORDER BY created_at DESC;`
+    )
```
Comment (nc-niamh-smith) [inline models/articles.model.js:20]: Really lovely sql query here - nice use of count and :: 🎉

**Review by nc-niamh-smith — COMMENTED** [review]
Lovely job here Mior - especially that tricky sql query! I've left you a comment about one of your expects, but it's just a suggestion :)

File: __tests__/integration.test.js (hunk 1 above)
Comment (miors) [reply __tests__/integration.test.js:94]: Thanks for the feedback @nc-niamh-smith , I have removed the expect keys 😊

**Pull request merged**